crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -738,9 +786,31 @@ where
                     }
 
                     self.visit_body(handler_body);
+
+                    // Each `except` block is mutually exclusive with all other `except` blocks.
+                    post_except_states.push(self.flow_snapshot());
+                    self.flow_restore(pre_except_state.clone());
```

Comment:
Nit. We could avoid this call for the last except handler because we restore the `post_try_block_state` on line 797 but I'm not sure if it's worth the complexity.